core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -208,7 +215,9 @@ public void removeLocation(long blockId, long workerId) {
 
   @Override
   public Iterator<Block> iterator() {
-    return RocksUtils.createIterator(db().newIterator(mBlockMetaColumn.get(), mIteratorOption),
+    // TODO(jiacheng): close the iterator when we iterate the BlockStore for backup
```

Comment:
this will be done in a separate PR as it takes a bit refactoring. Now the leak only happens in backing up.